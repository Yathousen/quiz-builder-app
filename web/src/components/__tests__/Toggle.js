import React from 'react';
import renderer from 'react-test-renderer';
import { Toggle } from '../';

it('Toggle Inactive', () => {
  expect(renderer.create(<Toggle name='Test Name' value={false} />).toJSON()).toMatchSnapshot();
});

it('Toggle Active', () => {
  expect(renderer.create(<Toggle name='Test Name' value={true} />).toJSON()).toMatchSnapshot();
});

it('Disabled Toggle', () => {
  expect(
    renderer.create(<Toggle name='Test Name' value={true} disabled />).toJSON(),
  ).toMatchSnapshot();
});
