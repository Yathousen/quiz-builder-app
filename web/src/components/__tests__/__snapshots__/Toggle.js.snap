// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabled Toggle 1`] = `
<div
  className="mt-4"
>
  <label
    className="block mb-2 text-sm font-medium text-gray-600 dark:text-gray-200"
    htmlFor="Test Name"
  >
    Test Name
  </label>
  <label
    className="mt-2 inline-flex items-center cursor-pointer"
    onClick={[Function]}
  >
    <span
      className="relative"
    >
      <span
        className="block w-10 h-6 bg-gray-200 rounded-full shadow-inner"
      />
      <span
        className="absolute block w-4 h-4 mt-1 ml-1 bg-white rounded-full shadow inset-y-0 left-0 focus-within:shadow-outline transition-transform duration-300 ease-in-out bg-purple-600 transform translate-x-full"
      />
    </span>
  </label>
</div>
`;

exports[`Toggle Active 1`] = `
<div
  className="mt-4"
>
  <label
    className="block mb-2 text-sm font-medium text-gray-600 dark:text-gray-200"
    htmlFor="Test Name"
  >
    Test Name
  </label>
  <label
    className="mt-2 inline-flex items-center cursor-pointer"
    onClick={[Function]}
  >
    <span
      className="relative"
    >
      <span
        className="block w-10 h-6 bg-gray-200 rounded-full shadow-inner"
      />
      <span
        className="absolute block w-4 h-4 mt-1 ml-1 bg-white rounded-full shadow inset-y-0 left-0 focus-within:shadow-outline transition-transform duration-300 ease-in-out bg-purple-600 transform translate-x-full"
      />
    </span>
  </label>
</div>
`;

exports[`Toggle Inactive 1`] = `
<div
  className="mt-4"
>
  <label
    className="block mb-2 text-sm font-medium text-gray-600 dark:text-gray-200"
    htmlFor="Test Name"
  >
    Test Name
  </label>
  <label
    className="mt-2 inline-flex items-center cursor-pointer"
    onClick={[Function]}
  >
    <span
      className="relative"
    >
      <span
        className="block w-10 h-6 bg-gray-200 rounded-full shadow-inner"
      />
      <span
        className="absolute block w-4 h-4 mt-1 ml-1 bg-white rounded-full shadow inset-y-0 left-0 focus-within:shadow-outline transition-transform duration-300 ease-in-out "
      />
    </span>
  </label>
</div>
`;
